### PR TITLE
refactor: resolve clippy warnings in wasm-dpp

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
Resolve clippy warnings in wasm-dpp

## What was done?
repo wide clippy warnings 326 -> 288

## How Has This Been Tested?
cargo t in root

## Breaking Changes
None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/wasm-dpp/src/data_contract/data_contract.rs [hunk 1]
```diff
@@ -153,7 +153,7 @@ impl DataContractWasm {
             .document_type_for_name(doc_type)
             .with_js_error()?;
 
-        let binary_paths_o = document_type.binary_paths();
+        let _unused_binary_paths_o = document_type.binary_paths();
```
Comment (shumkov) [inline packages/wasm-dpp/src/data_contract/data_contract.rs:156]: I think we just need to remove such unnecessary calls.

File: packages/wasm-dpp/src/data_contract_factory/data_contract_factory.rs [hunk 2]
```diff
@@ -32,7 +32,7 @@ impl DataContractFactoryWasm {
         protocol_version: u32,
         external_entropy_generator_arg: Option<ExternalEntropyGenerator>,
     ) -> DataContractFactoryWasm {
-        if let Some(external_entropy_generator) = external_entropy_generator_arg {
```
Comment (shumkov) [inline packages/wasm-dpp/src/data_contract_factory/data_contract_factory.rs:35]: This condition doesn't make any sense

File: packages/wasm-dpp/src/document/extended_document.rs [hunk 3]
```diff
@@ -363,6 +362,9 @@ impl ExtendedDocumentWasm {
 }
 
 impl ExtendedDocumentWasm {
+    #[allow(dead_code)]
+    #[deprecated(note = "This function is marked as unused.")]
+    #[allow(deprecated)]
```
Comment (shumkov) [inline packages/wasm-dpp/src/document/extended_document.rs:367]: Why do we allow deprecated code in this function?

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/wasm-dpp/src/document/extended_document.rs (hunk 3 above)
Comment (PastaPastaPasta) [reply packages/wasm-dpp/src/document/extended_document.rs:367]: https://github.com/dashpay/platform/pull/1756#discussion_r1517134639

File: packages/wasm-dpp/src/data_contract/data_contract.rs (hunk 1 above)
Comment (PastaPastaPasta) [reply packages/wasm-dpp/src/data_contract/data_contract.rs:156]: probably better to do in another PR maybe? I wanted to keep diff / potential for objection low. Simply renaming is best to achieve that for now and get rid of warning. 

Although if it's thought best to remove; I could without too much effort

File: packages/wasm-dpp/src/data_contract_factory/data_contract_factory.rs (hunk 2 above)
Comment (PastaPastaPasta) [reply packages/wasm-dpp/src/data_contract_factory/data_contract_factory.rs:35]: 🤷

Comment (shumkov) [reply packages/wasm-dpp/src/data_contract_factory/data_contract_factory.rs:35]: Just remove it

Comment (PastaPastaPasta) [reply packages/wasm-dpp/src/data_contract_factory/data_contract_factory.rs:35]: 73e1eb3664b652bd431ccb0bf56370ee81b7380e

File: packages/wasm-dpp/src/data_contract/data_contract.rs (hunk 1 above)
Comment (QuantumExplorer) [reply packages/wasm-dpp/src/data_contract/data_contract.rs:156]: remove

File: packages/wasm-dpp/src/data_contract/data_contract_facade.rs [hunk 4]
```diff
@@ -31,7 +31,7 @@ impl From<DataContractFacade> for DataContractFacadeWasm {
 pub struct DataContractFacadeWasm(pub(crate) Arc<DataContractFacade>);
 
 impl DataContractFacadeWasm {
-    pub fn new(protocol_version: u32, entropy_generator: ExternalEntropyGenerator) -> Self {
+    pub fn new(protocol_version: u32, _unused_entropy_generator: ExternalEntropyGenerator) -> Self {
```
Comment (QuantumExplorer) [inline packages/wasm-dpp/src/data_contract/data_contract_facade.rs:34]: This warning was useful and will now be missed. Best keep the warning here.

Comment (QuantumExplorer) [reply packages/wasm-dpp/src/data_contract/data_contract_facade.rs:34]: So revert

Comment (QuantumExplorer) [reply packages/wasm-dpp/src/data_contract/data_contract_facade.rs:34]: Or remove entropy_generator: ExternalEntropyGenerator

File: packages/wasm-dpp/src/identity/state_transition/identity_topup_transition/identity_topup_transition.rs [hunk 5]
```diff
@@ -67,7 +66,7 @@ impl IdentityTopUpTransitionWasm {
     pub fn set_asset_lock_proof(&mut self, asset_lock_proof: JsValue) -> Result<(), JsValue> {
         let asset_lock_proof = create_asset_lock_proof_from_wasm_instance(&asset_lock_proof)?;
 
-        self.0.set_asset_lock_proof(asset_lock_proof);
+        self.0.set_asset_lock_proof(asset_lock_proof).unwrap();
```
Comment (shumkov) [inline packages/wasm-dpp/src/identity/state_transition/identity_topup_transition/identity_topup_transition.rs:69]: You need to return error. Try ? or use map_err (see other methods) if it won't work automagically.

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/wasm-dpp/src/identity/state_transition/identity_topup_transition/identity_topup_transition.rs (hunk 5 above)
Comment (QuantumExplorer) [reply packages/wasm-dpp/src/identity/state_transition/identity_topup_transition/identity_topup_transition.rs:69]: Agreed

**Pull request merged**